pandas/core/frame.py

```diff
@@ -5746,9 +5751,11 @@ def _gotitem(self, key, ndim, subset=None):
         """
         if subset is None:
             subset = self
+        elif subset.ndim == 1:
```

Comment:
is this line actually hit in tests? return self._constructor here doesn't make sense

Reply:
Yes this is actually hit by the test case and code that was in place. A Series is a valid value for the subset parameter so I’m forcing it to a DataFrame or else the subsequent slice would fail. All for a better way if you think there’s one

Reply:
what operation actually hits this code?

Reply:
This line (which was changed to prevent the unlimited calls):

https://github.com/WillAyd/pandas/blob/4a24f734047d387ce242c36dba16eb69388a3ca1/pandas/core/base.py#L595

This would pass a Series before (unless column names were duplicated), though it never raised an error because of the code in `_got_item`. It would accept the Series and even use it in a condition, but would always just return a subset of itself...

Definitely convoluted - I think it was inadvertent before the way `_got_item` was implemented by DataFrame

Reply:
no the issue is why you are wrapping it with ``self._constructor`` which is a ``DataFrame`` here, then you are selecting it out again, just do

``return subset`` i think is enough?

Reply:
Hmm you're probably right with that - I suppose could just return immediately if `ndim == 1`. Will try locally and push if it works